modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -1173,3 +1116,67 @@ fun <A, E, B> IOOf<E, A>.bracketCase(release: (A, ExitCase2<E>) -> IOOf<E, Unit>
  */
 fun <E, A> IOOf<E, A>.guarantee(finalizer: IOOf<Nothing, Unit>): IO<E, A> =
   guaranteeCase { finalizer }
+
+/**
+ * Create a new [IO] that upon execution starts the receiver [IO] within a [Fiber] on [ctx].
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.*
+ * import arrow.fx.extensions.fx
+ * import kotlinx.coroutines.Dispatchers
+ *
+ * fun main(args: Array<String>) {
+ *   //sampleStart
+ *   val result = IO.fx {
+ *     val (join, cancel) = !IO.effect {
+ *       println("Hello from a fiber on ${Thread.currentThread().name}")
+ *     }.fork(Dispatchers.Default)
+ *   }
+ *
+ *   //sampleEnd
+ *   result.unsafeRunSync()
+ * }
+ * ```
+ *
+ * @receiver [IO] to execute on [ctx] within a new suspended [IO].
+ * @param ctx [CoroutineContext] to execute the source [IO] on.
+ * @return [IO] with suspended execution of source [IO] on context [ctx].
+ */
+fun <E, A> IOOf<E, A>.fork(ctx: CoroutineContext): IO<E, Fiber<IOPartialOf<E>, A>> =
+  IO.async { cb ->
+    val promise = UnsafePromise<E, A>()
+    // A new IOConnection, because its cancellation is now decoupled from our current one.
+    val conn = IOConnection()
+    IORunLoop.startCancelable(IOForkedStart(this, ctx), conn, promise::complete)
+    cb(IOResult.Success(IOFiber(promise, conn)))
+  }
+
+fun <A> IOOf<Nothing, A>.unsafeRunSync(): A =
+  fix().unsafeRunSyncEither()
+    .fold({
+      TODO()
+    }, ::identity)
+
+fun <A> IOOf<Nothing, A>.unsafeRunAsync(f: (Either<Throwable, A>) -> Unit): Unit =
```

Comment:
I seeeeee

Reply:
Ye, good idea to declare these on the IOOf to avoid confusion. The suffix Either doesn't thrill me, still, I don't have a better name for it

Reply:
> The suffix Either doesn't thrill me, still, I don't have a better name for it
So we can agree there :D Naming is so hard 😭

I attempted to overload with `@JvmName` but the concrete version always takes precedence so that doesn't work. Similar issue if we define both as extensions I think.